### PR TITLE
TEST: Increase coverage for shap/utils/image.py

### PR DESCRIPTION
**Pull request by Abhishek-Kumar-Rai5**
## Overview

Contributes to #3690  

## Summary

Adds comprehensive unit tests for `shap/utils/image.py` to significantly improve test coverage.

This PR targets previously untested functionality and validates behavior across filesystem operations, image processing, and resizing logic.

---

## Coverage Improvement

- Coverage increased from ~0% → ~95%

---

### Core Functions
- `is_empty` → directory existence and content checks  
- `make_dir` → directory creation and clearing behavior  
- `load_image` → valid and invalid image loading  
- `save_image` → file creation and output validation  
- `resize_image` → both resizing and non-resizing branches  

---

## Testing Approach

- Uses `pytest`
- Uses `tmp_path` for safe filesystem isolation
- Uses `numpy` arrays for image simulation
- Includes parameterized tests for extension validation
- Includes edge cases and failure scenarios

---

## Notes

- A bug in `make_dir` related to incorrect path concatenation is addressed in a separate PR #4501 
- This PR focuses purely on test coverage

---
## Checklist

- [x] All [pre-commit checks](https://pre-commit.com/#install) pass.
- [x] Unit tests added (if fixing a bug or adding a new feature)

## AI usage disclosure 
I used AI for the further test addition ideas and it helped in raising the coverage from 80 to 95%. Apart from it I used it to scan the image.py file and test file validation.

### CODE REVIEW

File: tests/utils/test_image.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+@pytest.mark.parametrize("filename", ["a.txt", "b", "c.pdf"])
+def test_check_valid_image_invalid(filename):
+    assert check_valid_image(filename) is None
+
+
+# ================== save + load ==================
+
+
+def test_save_and_load_image(tmp_path):
+    img = np.random.randint(0, 256, (100, 100, 3), dtype=np.uint8)
```
Comment (CloseChoice) [inline tests/utils/test_image.py:64]: pls fix a seed for all tests here.

**Review by CloseChoice — CHANGES_REQUESTED** [review]
